stk/calculators/optimization/optimizers.py

```diff
@@ -648,102 +650,501 @@ def optimize(self, mol, conformer=-1):
+
+        :class:`XTBConvergenceError`
+            If the optimization did not converge.
+
+        """
+        if output_file is None:
+            # No simulation has been run.
+            return True
+        # If convergence is achieved, then .xtboptok should exist.
+        if os.path.exists('.xtboptok'):
+            # Check for negative frequencies in output file if max_runs
+            # is > 1.
+            # Return True if there exists at least one.
+            if self.max_runs > 1:
+                return self._check_neg_frequencies(output_file)
+            else:
+                return False
+        elif os.path.exists('NOT_CONVERGED'):
+            raise XTBConvergenceError('Optimization not converged.')
+        else:
+            raise XTBOptimizerError('Optimization failed to complete')
+
+    def _run_xtb(self, xyz, out_file):
+        """
+        Runs GFN-xTB.
+
+        Parameters
+        ----------
+        xyz : :class:`str`
+            The name of the input structure ``.xyz`` file.
+
+        out_file : :class:`str`
+            The name of output file with xTB results.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        """
+
+        # Modify the memory limit.
+        if self.unlimited_memory:
+            memory = 'ulimit -s unlimited ;'
+        else:
+            memory = ''
+
+        # Set optimization level and type.
+        if self.calculate_hessian:
+            # Do optimization and check hessian.
+            optimization = f'--ohess {self.opt_level}'
+        else:
+            # Do optimization.
+            optimization = f'--opt {self.opt_level}'
+
+        if self.solvent is not None:
+            solvent = f'--gbsa {self.solvent} {self.solvent_grid}'
+        else:
+            solvent = ''
+
+        cmd = (
+            f'{memory} {self.xtb_path} {xyz} --gfn {self.gfn_version} '
+            f'{optimization} --parallel {self.num_cores} '
+            f'--etemp {self.electronic_temperature} '
+            f'{solvent} --chrg {self.charge} '
+            f'--uhf {self.num_unpaired_electrons}'
+        )
+
+        with open(out_file, 'w') as f:
+            # Note that sp.call will hold the program until completion
+            # of the calculation.
+            sp.call(
+                cmd,
+                stdin=sp.PIPE,
+                stdout=f,
+                stderr=sp.PIPE,
+                # Uses the shell if unlimited_memory is True to be run
+                # multiple commpands in one subprocess.
+                shell=self.unlimited_memory
+            )
+
+    def _run_optimizations(self, mol, conformer):
+        """
+        Run loop of optimizations of the molecule `mol` using xTB.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The molecule to be optimized.
+
+        conformer : :class:`int`
+            The conformer to use.
+
+        Returns
+        -------
+        None : :class:`NoneType`
```

Comment:
I belive this functions returns True or False

Reply:
please check the changes here